File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAbsoluteSum = function(nums) {
    	let highestPeak = 0;      
    	let deepestValley = 0;    
    	let currentClimb = 0;     
    	let currentDescent = 0;   
    
    	for (let number of nums) {
        	currentClimb = Math.max(number, currentClimb + number);
        	highestPeak = Math.max(highestPeak, currentClimb);
        
        	currentDescent = Math.min(number, currentDescent + number);
       	 	deepestValley = Math.min(deepestValley, currentDescent);
    }
    

   	return Math.max(Math.abs(highestPeak), Math.abs(deepestValley));
};

File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.py
class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        highest_peak = 0
        deepest_valley = 0
        current_climb = 0
        current_descent = 0
        for i in range(len(nums)):
            current_climb = max(nums[i], current_climb + nums[i])
            highest_peak = max(highest_peak, current_climb)
            
            current_descent = min(nums[i], current_descent + nums[i])
            deepest_valley = min(deepest_valley, current_descent)
        return max(abs(highest_peak), abs(deepest_valley))
        